docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -7,266 +7,200 @@
+'    and another where you get the registration report which requires some re-organization of calls.
+'    I think from a high-level there should be following three high-level calls either in
+'    ECC or ECCMAnager depending on above:
+'    - EnclaveCreate (key-gen & imprint of context, i.e., current create/binding flows
+'    - EnclaveProof: gets org endorsement-sig, creates report, then quote
+'      and, if epid, gets IAS report. 
+'    - EnclavePersist: returns sealed state (important for security that this is called only after
+'      EnclaveProof [which we might want to enforce using some state-machine?] )
+'  end note
+'  return ATTESTATION
 
 end
 
-group register attestation
+group Org-Enclave binding/certification (to be implemented)
+  Admin -> E_Manager ++: certify(CC_Id, Cert_Parameters)
+  E_Manager -> ECC1 ++: getCSR(Cert_Parameters)
+  ECC1 -> Enclave1 ++: getCSR(Cert_Parameters)
+  return csr /' ECC1 -> Enclave1 '/
+  return csr /' E_Manager -> ECC1 '/
+  E_Manager -> E_Manager : generate enclave certificate (certified by Org_1)
+  return certificate
+  note over Admin, E_Manager #FFaa00
+    -Org_1 takes "ownership" of this enclave,
+    -the certificate should be made available to the chaincode endorsement plugin,
+     so to include it in the responses (for eventual validation) 
+  end note
+end group
 
+group enclave registration
   activate Admin
+  Admin -> E_Manager ++: register
+  E_Manager -> ERCC1 ++: register Credentials
+  E_Manager -> ERCC2 : register Credentials
 
-  Admin -> ERCC1 ++: register (CC_NAME, CC_VER, MSP_ID_ORG_1, SIG_Org_1, ECC_S_PK, ECC_E_PK, ATTESTATION)
-  group verify and store
-
-    ERCC1 -> ERCC1 : check attestation signature & status
-    note left
-      - IAS verification key defined in ERCC definition
-      - status denotes the attestation result
-    end note
-
-    ERCC1 -> ERCC1 : Retrieve ECC definition for <CC_NAME, CC_VER>
+  ref over ERCC2
+    (same as ERCC1)
+  end ref
+  activate ERCC2
 
-    ERCC1 -> ERCC1 : Verify SIG_Org_1
+  ERCC1 -> ERCC1 : check Credentials.Evidence
+  ERCC1 -> ERCC1 : check Credentials.Attested_Data.CC_Parameters
 
-    ERCC1 -> ERCC1 : check consistency of quote (report-body)
-    note left
-      ECC_MR (mrenclave of ecc) is defined in ECC definition;
-      TLCC_MR (mrenclave of tlcc) is defined in ERCC definition
-      CH_Id is read from ledger or defined in ERCC definition
+  group check consistency of Credentials (TBD)
+    note over ERCC1 #FFaa00
+        The interaction with TLCC is not relevant. Rather, it is convenient to
+        access the chaincode definition.
+    end note
+    create "TLCC Peer1_Org1" as TLCC1 order 45
+    ERCC1 -> TLCC1  ++: approve consistency of Credentials
+    TLCC1 -> TLCC1    : check Credentials against chaincode definition
+    note over TLCC1 #FFaa00
+        This check ensures that:
+        1) the chaincode version (i.e., the MREnclave) matches the version of the
+            approved chaincode.
```

Comment:
(interesting point)
Not sure about this. I rather think we do not, here, and for multiple reasons:
1.  `TLCC_MREnclave` is hard-coded in ERCC, and already checked by ERCC on registration;
2.  ERCC is our root-of-trust for this value, since many participants have to agree on it
3. in a heterogeneous deployment, some peers may not have TLCC (and ERCC would check the chaincode definition by pulling it somehow from Fabric); still, the ERCC check is sufficient because it ensures that the chaincode will not establish a secure channel with a different TLCC -- if this is actually sound, it would be compelling for network with SGX-enabled designated FPC peers mixed with legacy Fabric peers.

Reply:
when i read it i saw the comment of everything which has to be verified during registration and verification of TLCC_MREnclave would be a crucial part.  Now whether that is done in ercc or tlcc that is an orthogonal question (and to some extent irrelevant from a protocol perspective where tlcc comes in only as the potential source of some information but not really from the usual 'tlcc-as-trusted-leger-for-ecc-in-hostile-environment' context^1 . Personally i think it is cleaner and easier to understand if all checks are done at a single place (and preferably ercc) but i guess that might be another of the 'agree-to-disagree' points. 

^1: given that we all agreed that for situations like these the tlcc enclave (which is really the heart of tlcc) is not involved, i think we should actually add a separate system-chaincode/component to make clear that this has nothing to do with the usual goal/functionality of tlcc ....

Reply:
* clarified the parameters that ercc checks,
* added a separate component and a comment to emphasize no need of enclave